kani-driver/src/assess/metadata.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use std::fs::File;
+use std::io::BufWriter;
+use std::path::Path;
+
+use anyhow::Result;
+use serde::{Deserialize, Serialize};
+
+use super::table_builder::TableBuilder;
+use super::table_failure_reasons::FailureReasonsTableRow;
+use super::table_promising_tests::PromisingTestsTableRow;
+use super::table_unsupported_features::UnsupportedFeaturesTableRow;
+use super::AssessArgs;
+
+/// The structure of `.kani-assess-metadata.json` files, which are emitted for each crate.
+/// This is not a stable interface.
+#[derive(Deserialize)]
+pub struct AssessMetadata {
+    pub unsupported_features: Vec<UnsupportedFeaturesTableRow>,
+    pub failure_reasons: Vec<FailureReasonsTableRow>,
+    pub promising_tests: Vec<PromisingTestsTableRow>,
+}
+
+/// This should be kept identical to [`AssessMetadata`], but lifetimes will differ.
```

Comment:
After writing up an explanation, I realized there might be a solution that lets me have just one, and sure enough, using `TableBuilder`, but adding `Serialize/Deserialize` implementations to it directly, worked! So that's cleaned up nicely.